File: playground/read_config/read_data.c
extern int load_file(char* name);
extern int find_value(int value);

int main(int argc, char* argv[]) {
    char* filename = argv[1];
    int res = load_file(filename);
    if (res != 0) {
        return res;
    }
    int val = find_value(7);
    printf("%i\n", val);
    return 0;
}

File: playground/read_config/read_data.py
import sys
from read_config.read_data_extension import find_value_c, load_file_c

data = []

def find_value(value):
    global data
    try:
        return data.index(value)
    except ValueError:
        return -1

def load_file(name):
    global data
    with open(name, "r") as f:
        data = [int(x) for x in f.readlines()]

def main():
    filename = sys.argv[1]
    load_file_c(filename)
    print(find_value_c(8496))
